lista3.sc

```diff
@@ -0,0 +1,162 @@
+//Jakub Kochański
+def reverse[A](list: List[A]): List[A] = {
+  @scala.annotation.tailrec
+  def reverseRec(accumulator: List[A])(newList: List[A]): List[A] = {
+    if (newList == Nil) accumulator
+    else reverseRec(newList.head :: accumulator)(newList.tail)
+  }
+
+  reverseRec(List())(list)
+}
+
+def createList(from: Int)(to: Int): List[Int] = {
+  @scala.annotation.tailrec
+  def createListRec(current: Int)(accum: List[Int]): List[Int] =
+    if (current <= to) createListRec(current + 1)(current :: accum)
+    else reverse(accum)
+
+  createListRec(from)(List())
+}
+
+//Zadanie 1
+val splitList: List[Int] => (List[Int], List[Int]) = listToSplit => {
+  def splitListRec: List[Int] => (List[Int], List[Int]) => (List[Int], List[Int]) = newList => (first, second) =>
+    if (newList == Nil) (first, second)
+    else if (newList.head % 2 == -1) splitListRec(newList.tail)(newList.head :: first, newList.head :: second)
+    else if (newList.head < 0) splitListRec(newList.tail)(newList.head :: first, second)
```

Comment:
jakby warunki zagnieździć to dając najpierw isNegative można by jeszcze zoptymalizować, ale i tak nie jest źle